spyder/config/user.py

```diff
@@ -447,14 +584,376 @@ def set(self, section, option, value, verbose=False, save=True):
+            self.reset_to_defaults(save=False)
+        else:
+            # TODO: Remove comment if tests pass ok
+            # self._update_defaults(defaults, old_version)
+            self._update_defaults(self.defaults, old_version)
+
+        if old_version and check_version(old_version, '44.1.0', '<'):
+            run_lines = to_text_string(self.get('ipython_console',
+                                                'startup/run_lines'))
+            if run_lines is not NoDefault:
+                run_lines = run_lines.replace(',', '; ')
+                self.set('ipython_console', 'startup/run_lines', run_lines)
+
+
+class MultiUserConfig(object):
+    """
+    Multi user config class based emulating the basic interface to UserConfig.
+
+    This class provides the same basic interface as UserConfig but allows
+    splitting the configuration sections and options among several files.
+
+    The `name` is now a `name_map` where the sections and options per file name
+    are defined.
+    """
+    DEFAULT_FILE_NAME = 'spyder'
+
+    def __init__(self, name_map, path, defaults=None, load=True, version=None,
+                 backup=False, raw_mode=False, remove_obsolete=False):
+        """Multi user config class based on UserConfig class."""
+        self._name_map = self._check_name_map(name_map)
+        self._path = path
+        self._defaults = defaults
+        self._load = load
+        self._version = version
+        self._backup = backup
+        self._raw_mode = 1 if raw_mode else 0
+        self._remove_obsolete = remove_obsolete
+
+        self._configs_map = {}
+        self._config_defaults_map = self._get_defaults_for_name_map(defaults,
+                                                                    name_map)
+        self._config_kwargs = {
+            'path': path,
+            'defaults': defaults,
+            'load': load,
+            'version': version,
+            'backup': backup,
+            'raw_mode': raw_mode,
+            'remove_obsolete': remove_obsolete,
+        }
+        for name in name_map:
+            defaults = self._config_defaults_map.get(name)
+            mod_kwargs = {
+                'name': name,
+                'defaults': defaults,
+            }
+            new_kwargs = self._config_kwargs.copy()
+            new_kwargs.update(mod_kwargs)
+            config_class = self.get_config_class()
+            self._configs_map[name] = config_class(**new_kwargs)
+
+    def _get_config(self, section, option):
+        """Get the correct configuration based on section and option."""
+        # Check the filemap first
+        name = self._get_name_from_map(section, option)
+        config_value = self._configs_map.get(name, None)
+
+        if config_value is None:
+            config_value = self._configs_map[self.DEFAULT_FILE_NAME]
+        return config_value
+
+    def _check_name_map(self, name_map):
+        """Check `name_map` follows the correct format."""
+        pass

```

Comment:
Why is there a pass here and a return below?

Reply:
This is a check for the name map to make sure  section/options are not repeated on several files. (updating now)